cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -105,6 +107,29 @@ neighbor_sample_impl(raft::handle_t const& handle,
   graph_view_t<vertex_t, edge_t, false, multi_gpu> modified_graph_view = graph_view;
   edge_masks_vector.reserve(num_edge_types);
 
+  label_t num_unique_labels = 0;
+
+  std::optional<rmm::device_uvector<label_t>> cp_starting_vertex_labels{std::nullopt};
```

Comment:
What is `cp`? `copy`? Once the follow-up if block is executed, this is no longer a copy of the original starting vertex labels. This stores unique starting vertex labels, and I think `cp_starting_vertex_labels` is a misnomer.

Reply:
Yes it is copy. 

> This stores unique starting vertex labels, and I think cp_starting_vertex_labels is a misnomer. 

the values in `cp_starting_vertex_labels` are not unique. I am only using this array to compute the number of unique labels

Reply:
We'd better reduce the scope of this variable.

You are using `num_unique_labels` or `cp_starting_vertex_labels` only after invoking `shuffle_and_organize_output`.

The input vertex labels are `const` and won't be modified in this function (even though you're updating `starting_vertices` to point different arrays).

No reason to pre-allocate memory here and also define this variable way before it is actually used. This goes against minimizing memory footprint & code readability.